lib/getData.py

```diff
@@ -0,0 +1,103 @@
+import os
+import fire
+import shutil
+
+import numpy as np
+import astroquery.mast as mast
+import astropy.io.fits as pyfits
+
+def getData (pi, target, pid, sciFil, imgFil, fileSearch='ima.fits', local='.', sortPath='.'):
+    """
+    Description
+    ----------
+    Downloads data from MAST for a given pi, target, and proposal id, and sorts into directories for WFC3.
+
+    Parameters
+    ----------
+    pi: string
+        Name of pi, wild cards are allowed
+    target: string
+        Name of target star
+    pid: string
+        Proposal id number.
+    sciFil: string
+        Filter used for science observations
+    imgFil: string
+        Filter used for direct images (0th order spectra)
+    fileSearch: string (optional)
+        Name subsearch for data files. Defaults to ima.fits
+    local: string (optional)
+        Directory to store mastDownload hierarchy of directories.
+    sortPath: string (optional)
+        Directory to store sorted files for WFC3
+
+    Returns
+    -------
+    None, but makes lots of files.
+
+    Warning
+    -------
+    This was intended to download scan files, and so hard coded to sort for SCAN_TYP='C'.
+    Will duplicate files so you don't redownload data. Watch your hard disk space. 
+
+    """
+    local = os.path.expanduser(local)
+    sortPath = os.path.expanduser(sortPath)
+
+    # find all observations associated with a target in a given pi's proposal
+    obs_table = mast.Observations.query_criteria(proposal_pi=pi, target_name=target, proposal_id=int(pid))
+
+    # get a list of associated data products
+    data_products_by_obs = mast.Observations.get_product_list(obs_table)
+
+    # search for only the ima fits files
+    indices = [row for row in np.arange(0, len(data_products_by_obs)) if fileSearch in data_products_by_obs['productFilename'][row]]
+
+    # download only these data products and don't redownload if you already have them
+    print(f'Downloading {len(indices)} Files:')
+    manifest = mast.Observations.download_products(data_products_by_obs[indices], extension="fits", cache=True, download_dir=local)
+
+    # sort through headers looking for image files, cals, and science files
+    cals = []
+    images = []
+    science = []
+    for file in manifest['Local Path']:
+        with pyfits.open(file) as hdul:
+            filt = hdul[0].header['FILTER']
+            scan = hdul[0].header['SCAN_TYP']
+        if filt == sciFil and scan == 'C':
+            science.append(file)
+        elif filt == imgFil:
+            images.append(file)
+        if scan != 'C':
```

Comment:
Should this be `elif`?  Do we want all the image files included in the `cals` list?

Reply:
Per slack: we want the image and the G141N put in cal. So image files are both in the list of `images` and in the list of`cals`, which will be moved to the `cals` directory. But our G141N example is only in `cals` and not in the `images` list. The prettiest logic is probably:
```
if (filt == sciFil) and (scan == 'C'):
    science.append(file)
elif (filt == imgFil) and (scan == 'N'):
    images.append(file)
    cals.append(file)
else:
    cals.append(file)
```

Reply:
Got it.  I understand what's going on now.